src/plugins/intel_cpu/src/emitters/x64/jit_load_store_emitters.cpp

```diff
@@ -97,6 +98,26 @@ size_t jit_load_emitter::aux_gprs_count() const {
 }
 
 void jit_load_emitter::emit_impl(const std::vector<size_t> &in_idxs, const std::vector<size_t> &out_idxs) const {
+    if (m_snippets_segfault_detector) {
+        // save runtime debug info
+        h->push(h->r15);
+        Xbyak::Label label_set_current;
+        h->mov(h->r15, reinterpret_cast<size_t>(&start_address));
+        h->cmp(h->qword[h->r15], 0);
+        h->jne(label_set_current);
+        h->mov(h->qword[h->r15], Xbyak::Reg64(in_idxs[0]));
+        h->L(label_set_current);
+        {
+            h->mov(h->r15, reinterpret_cast<size_t>(&current_address));
+            h->mov(h->qword[h->r15], Xbyak::Reg64(in_idxs[0]));
+
+            // iteration++
+            h->mov(h->r15, reinterpret_cast<size_t>(&iteration));
+            h->add(h->qword[h->r15], 0x01);
+        }
+        h->pop(h->r15);
+    }
```

Comment:
Can we totally move all debug info building code in Snippets emitters? We anyway use segfault handler only for Snippets. If I am not mistaken, this will allow us to avoid additional field `snippets_direct_emitter` introduction.

Reply:
Removed all segfault debug info related from jit_load/store_emitters.
"snippets_direct_emitter" is used in emit_code() of  base class jit_emitter. For jit_load/store_emitters, it's false to not trigger build_debug_info to override current emitter. It seems like "m_snippets_segfault_detector" could be reused for this goal and not introduce new snippets_direct_emitter. done in [a2826fb](https://github.com/openvinotoolkit/openvino/pull/19935/commits/a2826fb2cac577be511f9f6f681abb5b0830a6d7)